process/economics/builtInFunctionsCost.go

```diff
@@ -0,0 +1,173 @@
+package economics
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/mitchellh/mapstructure"
+)
+
+// ArgsBuiltInFunctionCost holds all components that are needed to create a new instance of builtInFunctionsCost
+type ArgsBuiltInFunctionCost struct {
+	GasSchedule core.GasScheduleNotifier
+	ArgsParser  process.ArgumentsParser
+}
+
+type builtInFunctionsCost struct {
+	gasConfig               *process.GasCost
+	specialBuiltInFunctions map[string]struct{}
+	argsParser              process.ArgumentsParser
+}
+
+// NewBuiltInFunctionsCost will create a new instance of builtInFunctionsCost
+func NewBuiltInFunctionsCost(args *ArgsBuiltInFunctionCost) (*builtInFunctionsCost, error) {
+	if args == nil {
+		return nil, process.ErrNilArgsBuiltInFunctionsConstHandler
+	}
+	if check.IfNil(args.ArgsParser) {
+		return nil, process.ErrNilArgumentParser
+	}
+	if check.IfNil(args.GasSchedule) {
+		return nil, process.ErrNilGasSchedule
+	}
+
+	bs := &builtInFunctionsCost{
+		argsParser: args.ArgsParser,
+	}
+
+	bs.initSpecialBuiltInFunctionCostMap()
+
+	var err error
+	bs.gasConfig, err = createGasConfig(args.GasSchedule.LatestGasSchedule())
+	if err != nil {
+		return nil, err
+	}
+
+	args.GasSchedule.RegisterNotifyHandler(bs)
+
+	return bs, nil
+}
+
+func (bc *builtInFunctionsCost) initSpecialBuiltInFunctionCostMap() {
```

Comment:
rename?
"special" is too generic, what kind of special?

Reply:
removed special